qt/python/mantidqt/widgets/superplot/presenter.py

```diff
@@ -53,6 +58,30 @@ def __init__(self, canvas, parent=None):
         self._update_spectrum_slider()
         self._update_hold_button()
 
+    def set_bin_mode(self, state):
+        """
+        Set the plot mode.
+
+        Args:
+            state (bool): if true, bin mode is set, if false, spectrum mode
+        """
+        if state:
+            self._view.set_mode(self.BIN_MODE_TEXT)
+        else:
+            self._view.set_mode(self.SPECTRUM_MODE_TEXT)
+        self._update_spectrum_slider()
+        self._update_plot()
+
+    def enable_error_bars(self, state):
+        """
+        Enable/disable error bars in plot.
+
+        Args:
+            state (bool): if True, the error bars will be on
+        """
+        self._error_bars = state
+        self._update_plot()
```

Comment:
These new methods are missing tests. 
We should especially check `self._update_plot` is called (just substitute the method with a mock within the test) as that's very easy to remove, introducing visual bugs that are only found with manual testing.

Reply:
I added the missing tests.